usr/sbin/p11sak/p11sak.c

```diff
@@ -913,6 +924,81 @@ static void set_bool_attr_from_string(CK_ATTRIBUTE *attr, char attr_char)
     }
 }
 
+/**
+ * @brief Set the custom attrs from string object
+ * 
+ * @param attr The attribute array
+ * @param attr_char The string from the user
+ * @param count Index of current attribute in the attribute array
+ */
+static CK_RV set_custom_attrs_from_string(char *attr_char, CK_ATTRIBUTE *attr_array, CK_ULONG *count) 
+{
+    CK_RV rc = CKR_OK;
+    char *attr = NULL;
+    char *separator = ",";
+
+    // iterate over the single custom attributes
+    attr = strtok(attr_char, separator);
+    while (attr != NULL && *count < 128)
+    {
+        int index = -1;
+        unsigned long id = 0;
+        char *key = NULL;
+        char *value = NULL;
+
+        // parse attribute id and value from single attribute
+        key = strtok(attr, "=");
```

Comment:
does this work? This will 'cancel' the above strtok and setup a new one. You can not mix 2 instances of tokenizing with the strtok call. strok internally uses a static state variable, which will get overwritten by the second strtok instance. 
Use strtok_r() instead to have 2 tokenizing instances concurrently.

Reply:
or use strchr to look for the '=' character and then set the the found character to 0 to split the string into 2 parts.